packages/ses/src/make-hardener.js

```diff
@@ -123,28 +123,41 @@ const freezeTypedArray = array => {
 };
 
 /**
- * Create a `harden` function.
- *
- * @returns {Harden}
+ * @typedef {object} HardenerKit
+ * @property {import('../types.js').Harden} harden
+ * @property {import('../types.js').IsHardened} isHardened
+ * @property {import('../types.js').Harden} hardenIntrinsics
  */
-export const makeHardener = () => {
-  // Use a native hardener if possible.
-  if (typeof globalThis.harden === 'function') {
-    const safeHarden = globalThis.harden;
-    return safeHarden;
-  }
 
+/**
+ * Create a HardenerKit.
+ *
+ * @returns {HardenerKit}
+ */
+export const makeHardenerKit = () => {
   const hardened = new WeakSet();
+  const hardenedAtLockdown = new WeakSet();
+  let hardenedAtLockdownSize = 0;
+  let intrinsicsWereHardened = false;
+
+  /** @param {any} value */
+  function markHardened(value) {
+    weaksetAdd(hardened, value);
+  }
 
-  const { harden } = {
-    /**
-     * @template T
-     * @param {T} root
-     * @returns {T}
-     */
+  /** @type {HardenerKit} */
+  const { harden, isHardened, hardenIntrinsics } = {
+    isHardened(value) {
+      return weaksetHas(hardened, value);
+    },
+    hardenIntrinsics(root) {
+      intrinsicsWereHardened = true;
+      return harden(root);
```

Comment:
Can `harden` throw? Might need a`finally` here.

Reply:
Yes it can, and no need for finally. The goal for now is to leave harden permanently broken. We can discuss whether we want this situation to be resolvable somehow.